core/chains/evm/client/errors.go

```diff
@@ -153,14 +155,16 @@ var arbitrumFatal = regexp.MustCompile(`(: |^)(invalid message format|forbidden
 var arbitrum = ClientErrors{
 	// TODO: Arbitrum returns this in case of low or high nonce. Update this when Arbitrum fix it
 	// Archived ticket: story/16801/add-full-support-for-incorrect-nonce-on-arbitrum
-	NonceTooLow:           regexp.MustCompile(`(: |^)invalid transaction nonce$|(: |^)nonce too low(:|$)`),
-	NonceTooHigh:          regexp.MustCompile(`(: |^)nonce too high(:|$)`),
-	TerminallyUnderpriced: regexp.MustCompile(`(: |^)gas price too low$`),
-	InsufficientEth:       regexp.MustCompile(`(: |^)(not enough funds for gas|insufficient funds for gas \* price \+ value)`),
-	Fatal:                 arbitrumFatal,
-	L2FeeTooLow:           regexp.MustCompile(`(: |^)max fee per gas less than block base fee(:|$)`),
-	L2Full:                regexp.MustCompile(`(: |^)(queue full|sequencer pending tx pool full, please try again)(:|$)`),
-	ServiceUnavailable:    regexp.MustCompile(`(: |^)502 Bad Gateway: [\s\S]*$|network is unreachable|i/o timeout`),
+	NonceTooLow:                   regexp.MustCompile(`(: |^)invalid transaction nonce$|(: |^)nonce too low(:|$)`),
+	NonceTooHigh:                  regexp.MustCompile(`(: |^)nonce too high(:|$)`),
+	TerminallyUnderpriced:         regexp.MustCompile(`(: |^)gas price too low$`),
+	InsufficientEth:               regexp.MustCompile(`(: |^)(not enough funds for gas|insufficient funds for gas \* price \+ value)`),
+	Fatal:                         arbitrumFatal,
+	L2FeeTooLow:                   regexp.MustCompile(`(: |^)max fee per gas less than block base fee(:|$)`),
+	L2Full:                        regexp.MustCompile(`(: |^)(queue full|sequencer pending tx pool full, please try again)(:|$)`),
+	ServiceUnavailable:            regexp.MustCompile(`(: |^)502 Bad Gateway: [\s\S]*$|network is unreachable|i/o timeout`),
+	ServiceTemporarilyUnavailable: regexp.MustCompile(`(: |^)503 Service Temporarily Unavailable(:|$)`),
+	ServiceTimeout:                regexp.MustCompile(`(: |^)408 Request Timeout(:|$)`),
```

Comment:
This is fine here, but I think we should also add the same regexp to the one defined in lines 667 - 670 further down in this file:
```
// Linkpool, Blockdaemon, and Chainstack all return "request timed out" if the log results are too large for them to process
var defaultClient = ClientErrors{
	TooManyResults: regexp.MustCompile(`request timed out`),
}
```
This is used to detect rpc service timeouts in the case where it's a request for logs... I assume arbitrum would  responds the same way (`408 Request Timeout`) in that situation. So maybe just adding it to the string with an `|...`?

Reply:
(I also, wonder if the `request timed out` that Linkpool, Blockdaemon, & Chainstack return in the case of logs is something they can also return in the case of a submitted tx... may be  unlikely if it always just accepts the tx immediately and returns, rather than blocking on anything)

Reply:
I don't know for sure if the RPC would have that result for `TooManyResults`. I'd rather only add that if it comes up so we know for sure.

Reply:
If it can return `408 Request Timeout` for submitting a tx, then I'm certain it can for fetching logs... which is typically a much longer operation, so more likely to time out.   (The other way around, I'd be less confident about.)

All of the rpc servers have some way of reporting a request timeout. I couldn't check every single rpc provider on every chain while adding this error handling so I guess arbitrum must have been one of the ones I missed.

There is no potential downside of using "request timed out|408 Request Timeout" instead of `request timed out" for that reg ex because if it doesn't return 408 Request Timeout it won't change any behavior and if it does, it's clear from the message that the request must have timed out.

Reply:
If you just want to get this merged I can add it myself in a separate PR, but I do think this belongs there.